src/main/java/com/smartcar/sdk/Vehicle.java

```diff
@@ -141,10 +142,12 @@ public String vin() throws SmartcarException {
    */
   public boolean hasPermissions(String permission) throws SmartcarException {
     try {
-      List<String> vehiclePermissions = Arrays.asList(this.permissions());
+      if (this.permissions == null) {
```

Comment:
What about doing this in the `permissions()` method instead of here. So that even if someone calls `vehicle.permissions()` directly, it is cached.